performance_manager/lib/l2_headways.py

```diff
@@ -0,0 +1,383 @@
+        TempHeadways.metadata.tables[TempHeadways.__tablename__]
+        .insert()
+        .from_select(
+            insert_cols,
+            headways_final_query,
+        )
+    )
+
+    db_manager.truncate_table(TempHeadways)
+    result = db_manager.execute(insert_query)
+
+    process_logger.add_metadata(temp_rows_inserted=result.rowcount)
+    process_logger.log_complete()
+
+    return result.rowcount
+
+
+def update_headways_from_temp(db_manager: DatabaseManager) -> None:
+    """
+    update headways tables with UPDATE from SELECT of TempHeadways table
+    """
+    process_logger = ProcessLogger("l2_headways_update")
+    process_logger.log_start()
+    update_query = (
+        Headways.metadata.tables[Headways.__tablename__]
+        .update()
+        .values(
+            fk_trip_event_hash=TempHeadways.fk_trip_event_hash,
+            fk_tu_stopped_event=TempHeadways.fk_tu_stopped_event,
+            fk_vp_stopped_event=TempHeadways.fk_vp_stopped_event,
+            headway_seconds=TempHeadways.headway_seconds,
+        )
+        .where(Headways.fk_trip_event_hash == TempHeadways.fk_trip_event_hash)
+    )
+
+    result = db_manager.execute(update_query)
+
+    process_logger.add_metadata(rows_updated=result.rowcount)
+    process_logger.log_complete()
+
+
+def insert_new_headways(db_manager: DatabaseManager) -> None:
+    """
+    insert new headways into Headways table with INSERT from SELECT of
+    TempHeadways table
+    """
+    insert_cols = [
+        "fk_vp_stopped_event",
+        "fk_tu_stopped_event",
+        "fk_trip_event_hash",
+        "headway_seconds",
+    ]
+
+    process_logger = ProcessLogger("l2_headways_insert")
+    process_logger.log_start()
+    insert_query = (
+        Headways.metadata.tables[Headways.__tablename__]
+        .insert()
+        .from_select(
+            insert_cols,
+            sa.select(
+                TempHeadways.fk_vp_stopped_event,
+                TempHeadways.fk_tu_stopped_event,
+                TempHeadways.fk_trip_event_hash,
+                TempHeadways.headway_seconds,
+            )
+            .join(
+                Headways,
+                Headways.fk_trip_event_hash == TempHeadways.fk_trip_event_hash,
+                isouter=True,
+            )
+            .where((Headways.fk_trip_event_hash.is_(None))),
+        )
+    )
+    result = db_manager.execute(insert_query)
+
+    process_logger.add_metadata(rows_inserted=result.rowcount)
+    process_logger.log_complete()
+
+
+def process_headways(db_manager: DatabaseManager) -> None:
+    """
+    process new events from L1 FullTripEvents table and insert into
+    Headways table
+
+    """
+    process_logger = ProcessLogger("process_l2_headways")
+    process_logger.log_start()
+    try:
+        temp_table_rows = load_temp_headways(db_manager)
+
+        # gate to check for no records needing update / insert
+        if temp_table_rows == 0:
+            process_logger.log_complete()
+            return
```

Comment:
this seems like a funny way to do this, but i follow it. i would have guessed there was a bool on result for it.

Reply:
Yeah, since we aren't doing anything with the `int` value being returned (other than checking if it's 0 or not) a bool makes more sense, I'll make that change